frontend/utilities/url/index.ts

```diff
@@ -39,3 +39,75 @@ export const buildQueryStringFromParams = (queryParams: QueryParams) => {
   }
   return queryString;
 };
+
```

Comment:
Instead of having all the separate params with overlapping logic checks, I think something like this could be a nice clean way to approach it:

```ts
const reconcileMutuallyExclusiveHostParams = (
  label?: string,
  policyId?: number,
  policyResponse?: string,
  mdmId?: string,
  mdmStatus?: string,
  softwareId?: number,
  operatingSystemId?: number
): Record<string, unknown> => {
  if (label) {
    return {};
  }

  switch (true) {
    case !!policyId:
      return {
        policy_id: policyId,
        policy_response: policyResponse,
      };
    case !!mdmId:
      return {
        mdm_id: mdmId,
        mdm_status: mdmStatus,
      };
    case !!softwareId:
      return {
        software_id: softwareId,
      };
    case !!operatingSystemId:
      return {
        operating_system_id: operatingSystemId,
      };
    default:
      return {};
  }
};
```

Then you can just do this when you need to construct the queryParams in the API methods:

const queryParams = {
  page,
  per_page: perPage,
  query: globalFilter,
  team_id: teamId,
  device_mapping,
  ...getSortParams(sortOptions),
  ...getStatusParam(selectedLabels),
  ...reconcileMutuallyExclusiveParams(
    label,
    policyId,
    softwareId,
    mdmId,
    mdmEnrollmentStatus
  ),
};
```

Caveat: I probably didn't catch everything in the cases and haven't dev tested. This would definitely be a prime candidate for unit tests. But I think in the long run this would be a more maintainable approach by having all the mutually exclusive params in one place and hopefully easier to read :)